File: src/Tests/Behavioral/SpreadOperator/SpreadOperator.cs
namespace go;

using fmt = fmt_package;
using ꓸꓸꓸnint = System.Span<nint>;

partial class main_package {

internal static nint sum(params ꓸꓸꓸnint numsʗp) {
    var nums = numsʗp.slice();

    nint total = 0;
    foreach (var (_, n) in nums) {
        total += n;
    }
    return total;
}

internal static void Main() {
    var values = new nint[]{1, 2, 3}.slice();
    fmt.Println(sum(values.ꓸꓸꓸ));
    fmt.Println(sum(1, 2, 3));
}

} // end main_package

File: src/Tests/Behavioral/SpreadOperator/SpreadOperator.go
package main

import "fmt"

func sum(nums ...int) int {
	total := 0
	for _, n := range nums {
		total += n
	}
	return total
}

func main() {
	values := []int{1, 2, 3}

	// Unpacks to: sum(1, 2, 3)
	fmt.Println(sum(values...))

	// Variadic params
	fmt.Println(sum(1, 2, 3))
}
